tests/snuba/rules/conditions/test_event_frequency.py

```diff
@@ -277,77 +388,101 @@ def _run_test(self, minutes, data, passes, add_events=False):
-            event_data = deepcopy(data)
-            event_data["user"] = {"id": uuid4().hex}
-            self.store_event(
-                data=event_data,
-                project_id=self.project.id,
-            )
+            environment, _ = Environment.objects.get_or_create(name=environment)
+
+        for i in range(count):
+            if perf:
+                self.store_transaction(
+                    environment=environment,
+                    project_id=self.project.id,
+                    user_id=str(i),
+                    fingerprint=event.data["fingerprint"],
+                    timestamp=timestamp if pass_txn_timestamp else None,
+                )
+            else:
+                event_data = deepcopy(data)
+                event_data["user"] = {"id": uuid4().hex}
+                self.store_event(
+                    data=event_data,
+                    project_id=self.project.id,
+                )
 
     @patch("sentry.rules.conditions.event_frequency.MIN_SESSIONS_TO_FIRE", 1)
     def test_five_minutes_with_events(self):
         self._make_sessions(60)
         data = {"interval": "5m", "value": 39}
         self._run_test(data=data, minutes=5, passes=True, add_events=True)
+        self._run_test(data=data, minutes=5, passes=True, add_events=True, perf=True)
         data = {"interval": "5m", "value": 41}
         self._run_test(data=data, minutes=5, passes=False)
+        self._run_test(data=data, minutes=5, passes=False, perf=True)
 
     @patch("sentry.rules.conditions.event_frequency.MIN_SESSIONS_TO_FIRE", 1)
     def test_ten_minutes_with_events(self):
         self._make_sessions(60)
         data = {"interval": "10m", "value": 49}
         self._run_test(data=data, minutes=10, passes=True, add_events=True)
+        self._run_test(data=data, minutes=10, passes=True, add_events=True, perf=True)
         data = {"interval": "10m", "value": 51}
         self._run_test(data=data, minutes=10, passes=False)
+        self._run_test(data=data, minutes=10, passes=False, perf=True)
 
     @patch("sentry.rules.conditions.event_frequency.MIN_SESSIONS_TO_FIRE", 1)
     def test_thirty_minutes_with_events(self):
         self._make_sessions(60)
         data = {"interval": "30m", "value": 49}
         self._run_test(data=data, minutes=30, passes=True, add_events=True)
+        self._run_test(data=data, minutes=30, passes=True, add_events=True, perf=True)
         data = {"interval": "30m", "value": 51}
         self._run_test(data=data, minutes=30, passes=False)
+        self._run_test(data=data, minutes=30, passes=False, perf=True)
 
     @patch("sentry.rules.conditions.event_frequency.MIN_SESSIONS_TO_FIRE", 1)
     def test_one_hour_with_events(self):
         self._make_sessions(60)
         data = {"interval": "1h", "value": 49}
         self._run_test(data=data, minutes=60, add_events=True, passes=True)
+        self._run_test(data=data, minutes=60, add_events=True, passes=True, perf=True)
```

Comment:
I think that adding tests like this won't really scale well going forward as we add more issue types. It would be better to have a base test class that we subclass, like:

```
class EventFrequencyPercentConditionTestCase(...):
    def store_event():
        # Should create the event specific to this issue type
        raise NotImplementedError()

class ErrorEventFrequencyPercentConditionTestCase(EventFrequencyPercentConditionTestCase):
    def store_event(...):
        # create error events

class PerformanceEventFrequencyPercentConditionTestCase(EventFrequencyPercentConditionTestCase):
    def store_event(...):
        # create transaction events

```
Naming is probably not great, but something along these lines. That way `incr` can just make use of this generic way to create events and we minimize duplication

We should be able to apply this same general concept to all the tests here.